### PR TITLE
feature: added working hours

### PR DESCRIPTION
**Pull request by ramshty**
# Pull Request: Add Working Hours Feature to Worker Profile

## Summary
This PR adds the ability for workers to define their **working hours** directly within their profile. This feature enhances profile customization and allows workers to clearly communicate their availability to clients.

### Backend
- **Model Update:**
  - Added `workingHours` field to the `WorkerProfile` model:
    - Accepts a pair (`start`, `end`) representing the worker's daily availability.
- **Firestore Integration:**
  - Updated Firestore to support saving and retrieving the `workingHours` field.

### Frontend
- Added UI components to allow workers to set their working hours.
- Updated profile editing and viewing pages to include the working hours information.

## Benefits
- Workers can define their availability, improving clarity and client experience.
- Enhanced profile customization to suit individual preferences.

## Testing
- Verified that the `workingHours` field is correctly stored and retrieved from Firestore.
- Ensured the UI updates dynamically with the new working hours.
- Tested edge cases such as invalid time ranges and missing values.

## Checklist
- [x] Code changes implemented
- [x] UI updated to reflect changes
- [x] Unit and integration tests written and passed
- [x] Documentation updated (if necessary)

---
Please review and provide feedback.

### CODE REVIEW

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/TimeDropDownMenu.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+  // Dropdown Menu
+  DropdownMenu(
+      expanded = expanded,
+      onDismissRequest = { expanded = false },
+      modifier =
+          Modifier.heightIn(max = 240.dp) // Max height to fit ~6 rows
+              .semantics { testTag = "TimeDropdownMenu$tag" },
+      containerColor = colorScheme.surface) {
+        Box(modifier = Modifier.size(width = 80.dp * widthRatio.value, height = 300.dp)) {
```
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/TimeDropDownMenu.kt:99]: try to use height ratio instead of absolute values

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt [hunk 2]
```diff
@@ -190,17 +203,48 @@ fun ProfessionalInfoScreen(
...
+                  isStartTime = true, // This is the Start Time dropdown
+                  widthRatio = widthRatio,
+                  heightRatio = heightRatio,
+                  tag = "StartTime")
+            }
+            HorizontalDivider(
+                modifier =
+                    Modifier.weight(0.1f)
+                        .align(Alignment.CenterVertically)
+                        .padding(top = 20.dp, end = 12.dp))
```
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt:227]: try to not use absolut value

**Review by daferya — CHANGES_REQUESTED** [review]
You've written excellent tests that cover all the important cases and implemented exactly what we discussed during the meeting. Just be careful with absolute values and consider replacing them with relative ones.

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/UpgradeToWorkerScreen.kt [hunk 3]
```diff
@@ -127,9 +134,14 @@ fun BusinessScreen(
...
                 uid = accountId)
+        // Define the time formatter
+        val timeFormatter = DateTimeFormatter.ofPattern("HH:mm")
+        // Convert to string
+        val workingHoursString =
+            "${workingHours.value.first?.format(timeFormatter)} - ${workingHours.value.second?.format(timeFormatter)}"
         Log.d(
             "UpgradeToWorkerScreen",
-            "workerProfile: ${locationWorker.value.name} ${fieldOfWork.value} ${description.value} ${price.doubleValue} ${displayName.value} ${includedServices.value} ${addOnServices.value} ${tags.value} ${uploadedImageUrls[0]} ${if (uploadedImageUrls.size > 1) uploadedImageUrls[1] else ""} $accountId")
+            "workerProfile: ${locationWorker.value.name} ${fieldOfWork.value} ${description.value} ${price.doubleValue} ${displayName.value} ${includedServices.value} ${addOnServices.value} ${tags.value} ${uploadedImageUrls[0]} ${if (uploadedImageUrls.size > 1) uploadedImageUrls[1] else ""} $accountId $workingHoursString")
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/UpgradeToWorkerScreen.kt:144]: You can remove theses logs here 😄

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/UpgradeToWorkerScreen.kt [hunk 4]
```diff
@@ -116,6 +119,10 @@ fun BusinessScreen(
         // Make the announcement
         val workerProfile =
             WorkerProfile(
+                workingHours =
+                    if (workingHours.value.first == null || workingHours.value.second == null)
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/UpgradeToWorkerScreen.kt:123]: Checking if one of the value if null is good 👍🏼

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/personal/PersonalInfoScreen.kt [hunk 5]
```diff
@@ -456,7 +458,7 @@ fun PersonalInfoScreen(
       ) {
         QuickFixButton(
             buttonText = "Cancel",
-            onClickAction = {},
+            onClickAction = { navigationActions.goBack() },
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/personal/PersonalInfoScreen.kt:461]: Nice 👍🏼

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt (hunk 2 above)
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt:227]: Yup, absolute values here, could be avoided but it's ok

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt [hunk 6]
```diff
@@ -579,7 +624,7 @@ fun ProfessionalInfoScreen(
                                 .padding(start = 5.dp, end = 5.dp, bottom = 8.dp)) {
                           QuickFixButton(
                               buttonText = "Cancel",
-                              onClickAction = {},
+                              onClickAction = { navigationActions.goBack() },
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt:627]: 👍🏼

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt [hunk 7]
```diff
@@ -646,7 +693,7 @@ fun calculateMaxTextWidth(texts: List<String>, textStyle: TextStyle): Dp {
 suspend fun smoothScrollToItem(
     listState: LazyListState,
     targetIndex: Int,
-    durationMillis: Int = 5 // Total animation duration
+    durationMillis: Int = 1 // Total animation duration
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/ProfessionalInfoScreen.kt:696]: Maybe a bit too fast?

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/TimeDropDownMenu.kt (hunk 1 above)
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/TimeDropDownMenu.kt:99]: Again here, just add heightRatio.value

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/TimeDropDownMenu.kt [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+                    selectedTime.value = time // Update the selected LocalTime
+                    if (isStartTime && dependentTime.value != null && time >= dependentTime.value) {
+                      dependentTime.value = null // Reset the end time if invalid
+                    }
+                    expanded = false // Close the dropdown
+                  },
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .height(40.dp) // Fixed height per row
+                          .padding(horizontal = 8.dp)
```
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/profile/becomeWorker/views/professional/TimeDropDownMenu.kt:122]: Same here, use widthRatio.value

**Review by JitterKiller — APPROVED** [review]
Overall great PR!

Adding time slots for working hours is a good feature to have.
Just make sure to avoir using absolute values especially values like `300.dp` 😅

Other that LGTM
